File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        answer = 0
        n = len(mat)
        for i in range(n):
            answer += mat[i][i] + mat[i][n - i - 1]
            if i == n - i - 1:
                answer -= mat[i][i]
        return answer

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.go
func diagonalSum(mat [][]int) int {
    ans := 0
    for i := 0; i < len(mat); i++ {
        ans += mat[i][i]
        if len(mat) - i - 1 != i {
            ans += mat[i][len(mat) - i - 1]
        }
    }
    return ans
}